modules/apps/batch-engine/batch-engine-test/src/testIntegration/java/com/liferay/batch/engine/internal/exportimport/data/handler/test/BatchEnginePortletDataHandlerRegistryTest.java

```diff
@@ -0,0 +1,200 @@
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.search.filter.Filter;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.test.TestInfo;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.util.ClassUtil;
+import com.liferay.portal.kernel.util.HashMapDictionaryBuilder;
+import com.liferay.portal.kernel.util.MapUtil;
+import com.liferay.portal.odata.entity.EntityModel;
+import com.liferay.portal.test.rule.FeatureFlag;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.vulcan.batch.engine.VulcanBatchEngineTaskItemDelegate;
+import com.liferay.portal.vulcan.pagination.Page;
+import com.liferay.portal.vulcan.pagination.Pagination;
+
+import java.io.Serializable;
+
+import java.util.Collection;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Map;
+
+import javax.portlet.GenericPortlet;
+import javax.portlet.Portlet;
+
+import javax.ws.rs.core.UriInfo;
+
+import org.hamcrest.CoreMatchers;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+
+/**
+ * @author Alejandro Tardín
+ */
+@FeatureFlag("LPD-35914")
+@RunWith(Arquillian.class)
+public class BatchEnginePortletDataHandlerRegistryTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	@TestInfo("LPD-56301")
+	public void test() throws Exception {
+		String portletId = RandomTestUtil.randomString();
+
+		Assert.assertNull(
+			_portletDataHandlerProvider.provide(
+				TestPropsValues.getCompanyId(), portletId));
+
+		try (SafeCloseable safeCloseable1 = _registerServiceWithSafeCloseable(
+				Portlet.class,
+				new GenericPortlet() {
+				},
+				MapUtil.singletonDictionary("javax.portlet.name", portletId));
+			SafeCloseable safeCloseable2 = _registerServiceWithSafeCloseable(
+				VulcanBatchEngineTaskItemDelegate.class,
+				new VulcanBatchEngineTaskItemDelegate<>() {
+
+					@Override
+					public void create(
+						Collection<Object> items,
+						Map<String, Serializable> parameters) {
+					}
+
+					@Override
+					public void delete(
+						Collection<Object> items,
+						Map<String, Serializable> parameters) {
+					}
+
+					@Override
+					public EntityModel getEntityModel(
+						Map<String, List<String>> multivaluedMap) {
+
+						return null;
+					}
+
+					@Override
+					public String getPortletId() {
```

Comment:
I miss a test where the `getPortletId` method returns null

Reply:
The default implementation returns null, so I guess there is a chance to call that method if a portlet exists?